src/Domain/Orientation.cpp

```diff
@@ -57,12 +57,8 @@ template <size_t VolumeDim>
 Orientation<VolumeDim> Orientation<VolumeDim>::inverse() const noexcept {
   std::array<Direction<VolumeDim>, VolumeDim> result;
   for (size_t i = 0; i < VolumeDim; i++) {
-    const auto side =
-        mapped(Direction<VolumeDim>(i, Side::Upper)).side() == Side::Upper
-            ? Side::Upper
-            : Side::Lower;
     gsl::at(result, mapped(Direction<VolumeDim>(i, Side::Upper).dimension())) =
```

Comment:
you can replace `mapped(Direction<VolumeDim>(i, Side::Upper)` with `gsl::at(mapped_directions_, i)` here too